cmd/plugin_scaffold_test.go

```diff
@@ -0,0 +1,132 @@
+	assert.Contains(t, output, "test-gatewayd-plugin/.github/workflows/commits-signed.yaml")
+
+	pluginsConfig, err := os.ReadFile(
+		filepath.Join("plugins", "test-gatewayd-plugin", "gatewayd_plugin.yaml"))
+	require.NoError(t, err, "reading plugins config file should not return an error")
+
+	var localPluginsConfig map[string]interface{}
+	err = yamlv3.Unmarshal(pluginsConfig, &localPluginsConfig)
+	require.NoError(t, err, "unmarshalling yaml file should not return error")
+
+	tidy := exec.Command("go", "mod", "tidy")
+	tidy.Dir = filepath.Join("plugins", "test-gatewayd-plugin")
+	err = tidy.Run()
+	assert.NoError(t, err)
+
+	build := exec.Command("make", "build-dev")
+	build.Dir = filepath.Join("plugins", "test-gatewayd-plugin")
+	err = build.Run()
+	assert.NoError(t, err)
+
+	_, err = os.ReadFile(filepath.Join("plugins", "test-gatewayd-plugin", "test-gatewayd-plugin"))
+	require.NoError(t, err, "reading plugin binary file should not return an error")
+
+	pluginsList := cast.ToSlice(localPluginsConfig["plugins"])
+	plugin := cast.ToStringMap(pluginsList[0])
+	pluginsList[0] = plugin
+	plugin["localPath"] = filepath.Join("cmd", "plugins", "test-gatewayd-plugin", "test-gatewayd-plugin")
+	sum, err := checksum.SHA256sum(filepath.Join("plugins", "test-gatewayd-plugin", "test-gatewayd-plugin"))
+	require.NoError(t, err, "marshalling yaml file should not return error")
+	plugin["checksum"] = sum
+
+	pluginsList[0] = plugin
+	plugins := make(map[string]interface{})
+	plugins["plugins"] = pluginsList
+
+	updatedPluginConfig, err := yamlv3.Marshal(plugins)
+	require.NoError(t, err, "marshalling yaml file should not return error")
+
+	err = os.WriteFile(
+		filepath.Join("plugins", "test-gatewayd-plugin", "gatewayd_plugins.yaml"),
+		updatedPluginConfig, FilePermissions)
+	require.NoError(t, err, "writingh to yaml file should not return error")
+
+	pluginTestConfigFile := filepath.Join(
+		"plugins", "test-gatewayd-plugin", "gatewayd_plugins.yaml")
+
+	stopChan = make(chan struct{})
+
+	var waitGroup sync.WaitGroup
+
+	waitGroup.Add(1)
+	go func(waitGroup *sync.WaitGroup) {
+		// Test run command.
+		output, err := executeCommandC(
+			rootCmd, "run", "-c", globalTestConfigFile, "-p", pluginTestConfigFile)
+		require.NoError(t, err, "run command should not have returned an error")
+
+		// Print the output for debugging purposes.
+		runCmd.Print(output)
+		// Check if GatewayD started and stopped correctly.
+		assert.Contains(t, output, "GatewayD is running")
+		assert.Contains(t, output, "Stopped all servers")
+
+		waitGroup.Done()
+	}(&waitGroup)
+
+	waitGroup.Add(1)
+	go func(waitGroup *sync.WaitGroup) {
+		time.Sleep(waitBeforeStop)
+
+		StopGracefully(
+			context.Background(),
+			nil,
+			nil,
+			metricsServer,
+			nil,
+			loggers[config.Default],
+			servers,
+			stopChan,
+			nil,
+			nil,
+		)
+
+		waitGroup.Done()
+	}(&waitGroup)
+
+	waitGroup.Wait()
+
+	assert.NoError(t, os.RemoveAll("plugins"))
```

Comment:
if we use t.TempDir() we dont need this line

Reply:
Addressed in https://github.com/gatewayd-io/gatewayd/pull/613/commits/d7cf22171a6ca097fd3fcd32b5a889727c9bdf0d.